PiwikTracker.php

```diff
@@ -1169,41 +1168,21 @@ public function setIp($ip)
      */
     public function setUserId($userId)
     {
-        if ($userId === false) {
-            $this->setNewVisitorId();
-            return $this;
-        }
         if ($userId === '') {
             throw new Exception("User ID cannot be empty.");
         }
         $this->userId = $userId;
         return $this;
     }
 
-    /**
-     * Hash function used internally by Piwik to hash a User ID into the Visitor ID.
-     *
-     * Note: matches implementation of Tracker\Request->getUserIdHashed()
-     *
-     * @param $id
-     * @return string
-     */
-    public static function getUserIdHashed($id)
```

Comment:
@MichaelHeerklotz I wonder if we should maybe leave the method there and remove it in Matomo 4 to not break anything in case someone was using it? Not sure why it was made public.

Reply:
Good point. I have updated the commit.